pkg/controllers/scheduler/scheduler.go

```diff
@@ -268,23 +332,44 @@ func (s *Scheduler) reconcile(qualifiedName common.QualifiedName) (status worker
 					"object propagation policy %s not found",
 					policyKey.String(),
 				)
-				return worker.Result{Success: false, RequeueAfter: nil}
+				return nil, nil, nil, &worker.Result{Success: false, RequeueAfter: nil}
+			}
+			return nil, nil, nil, &worker.StatusError
+		}
+
+		profileName := policy.GetSpec().SchedulingProfile
+		if len(profileName) > 0 {
+			keyedLogger = keyedLogger.WithValues("profile", profileName)
+			schedulingProfile, err = s.schedulingProfileLister.Get(profileName)
+			if err != nil {
+				keyedLogger.Error(err, "Failed to get scheduling profile")
+				s.eventRecorder.Eventf(
+					fedObject,
+					corev1.EventTypeWarning,
+					EventReasonScheduleFederatedObject,
+					"failed to schedule object: %v",
+					fmt.Errorf("failed to get scheduling profile %s: %w", profileName, err),
+				)
+
+				if apierrors.IsNotFound(err) {
+					return nil, nil, nil, &worker.StatusAllOK
```

Comment:
should we return `&worker.Result{Success: false, RequeueAfter: nil}` instead of `&worker.StatusAllOK` to be consistent with line 335 above?

Reply:
Changed line 335 to return `&worker.StatusAllOK` as well. We should remove the `Success` field in `worker.Result` eventually.